l1-contracts/test/foundry/unit/concrete/updates/BaseZkSyncUpgrade.t.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {Test} from "forge-std/Test.sol";
+
+import {BaseZkSyncUpgrade, ProposedUpgrade} from "contracts/upgrades/BaseZkSyncUpgrade.sol";
+import {VerifierParams} from "contracts/state-transition/chain-interfaces/IVerifier.sol";
+import {MAX_NEW_FACTORY_DEPS} from "contracts/common/Config.sol";
+
+import {BaseUpgrade} from "./_SharedBaseUpgrade.t.sol";
+import {BaseUpgradeSetters} from "./_SharedBaseUpgradeSetters.t.sol";
+
+contract DummyBaseZkSyncUpgrade is BaseZkSyncUpgrade, BaseUpgradeSetters {}
+
+contract BaseZkSyncUpgradeTest is BaseUpgrade {
+    DummyBaseZkSyncUpgrade baseZkSyncUpgrade;
+
+    function setUp() public {
+        baseZkSyncUpgrade = new DummyBaseZkSyncUpgrade();
+
+        _prepereProposedUpgrade();
+
+        baseZkSyncUpgrade.setPriorityTxMaxGasLimit(1 ether);
+        baseZkSyncUpgrade.setPriorityTxMaxPubdata(1000000);
+    }
+
+    // Upgrade is not ready yet
+    function test_revertWhen_UpgradeIsNotReady() public {
+        proposedUpgrade.upgradeTimestamp = block.timestamp + 1;
```

Comment:
Maybe make it a fuzz test and bound `upgradeTimestamp` instead?

Reply:
https://github.com/matter-labs/era-contracts/pull/496#issuecomment-2137580336